scipy/stats/_binned_statistic.py

```diff
@@ -650,6 +643,27 @@ def binned_statistic_dd(sample, values, statistic='mean',
     return BinnedStatisticddResult(result, edges, binnumbers)
 
 
+def _calc_binned_statistic(Vdim, binnumbers, result, values, stat_func):
+    unique_bin_numbers = np.unique(binnumbers)
```

Comment:
the unique calculation is also done outside the loop, isn't it not needed here then?

Reply:
What do you mean? the unique calculation is only done here.

Reply:
I'm not sure why I thought that yesterday, sometimes the view in the github page is weird, maybe that is what happened. From what I see now I agree with your statement.